commcare_connect/opportunity/tables.py

```diff
@@ -947,3 +947,284 @@ def render_icons(self, record):
             justify_class,
             mark_safe(icons_html),
         )
+
+
+class UserInfoColumn(tables.Column):
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("orderable", True)
+        kwargs.setdefault("verbose_name", "Name")
+        super().__init__(*args, **kwargs)
+
+    def render(self, value):
+        return format_html(
+            """
+            <div class="flex flex-col items-start w-40">
+                <p class="text-sm text-slate-900">{}</p>
+                <p class="text-xs text-slate-400">{}</p>
+            </div>
+            """,
+            value.name,
+            value.username,
+        )
+
+
+class SuspendedIndicatorColumn(tables.Column):
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("orderable", False)
+        kwargs.setdefault(
+            "verbose_name", mark_safe('<div class="w-[40px]"><div class="w-4 h-2 bg-black rounded"></div></div>')
+        )
+        super().__init__(*args, **kwargs)
+
+    def render(self, value):
+        color_class = "negative-dark" if value else "positive-dark"
+        return format_html('<div class="w-10"><div class="w-4 h-2 rounded {}"></div></div>', color_class)
+
+
+def get_duration_min(total_seconds):
+    total_seconds = int(total_seconds)
+    minutes = (total_seconds // 60) % 60
+    hours = (total_seconds // 3600) % 24
+    days = total_seconds // 86400
+
+    parts = []
+    if days:
+        parts.append(f"{days} day{'s' if days != 1 else ''}")
+    if hours:
+        parts.append(f"{hours} hr")
+    if minutes or not parts:
+        parts.append(f"{minutes} min")
+
+    return " ".join(parts)
+
+
+class DurationColumn(tables.Column):
+    def render(self, value):
+        total_seconds = int(value.total_seconds() if isinstance(value, timedelta) else 0)
+        return get_duration_min(total_seconds)
+
+
+class WorkerStatusTable(tables.Table):
```

Comment:
Are these tables new or do they replace older ones?

Reply:
Good question, this particular one has a corresponding UserStatusTable, which is used for exports as well as the previous UI, I have kept it for now, since the existing opportunity detail page is still using it, but when we do the opportunity dashboard page, we can remove it.